04/app.js

```diff
@@ -4,31 +4,32 @@ import { createRoot } from 'react-dom/client';
 const root = createRoot(document.querySelector('#root'));
 
 class App extends React.Component {
-    state = { 
+    state = {
         firstName: '',
         lastName: '',
         searchQuery: '',
         users: ['Jan Kowalski', 'Michał Nowak'],
     }
 
     renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
+        const { users, searchQuery } = this.state;
+        const filteredUsers = users.filter(item => item.toLowerCase().includes(searchQuery))
```

Comment:
Dla pewności `searchQuery` też powinno być `.toLowetCase()` - inaczej gdy użytkownik napisze coś z wielkiej litery to nie zadziała.